Homework_8_Java_Classes_Objects_Fields_Methods/MyAmazingRPG/src/MyAmazingRPG.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * Механика игры:
+ * 1. Пока есть живые герои и враги, случайным образом выбирается герой и враг для сражения.
+ * 2. Выводится информация о выбранных персонажах.
+ * 3. Проходит битва между выбранными персонажами, пока один из них не умрет. Во время битвы каждый персонаж поочередно
+ * атакует другого, и после каждой атаки выводится текущее состояние персонажей.
+ * 4. Значение здоровья у победившего персонажа не восстанавливается (т.е. в следующем бою этот персонаж будет иметь
+ * столько здоровья, сколько у него осталось после предыдущей битвы).
+ * 5. Если герои побеждают, то последний герой должен сразится с супермонстром.
+ * Результаты битвы:
+ * 1. После завершения битвы проверяется, кто победил. Проигравший удаляется из списка героев или врагов.
+ * 2. Если герои или враги закончились, выводится соответствующее сообщение о победе или поражении.
+ * 3. После битвы с супермонстром выводится сообщение о победе или поражении.
+ */
+public class MyAmazingRPG {
+    public static void main(String[] args) {
+        //Создаю список героев
+        ArrayList<Character> heroes = new ArrayList<>();
+        heroes.add(new Character("Code Warrior", 120, 30, 15));
+        heroes.add(new Character("Bug Hunter", 100, 20, 10));
+        heroes.add(new Character("Server Savior", 90, 25, 20));
+        heroes.add(new Character("Pixel Mage", 90, 20, 10));
+        heroes.add(new Character("Task Master", 80, 30, 50));
+
+        //Создаю список врагов
+        ArrayList<Character> enemies = new ArrayList<>();
+        enemies.add(new Character("Junior Bug", 80, 10, 5));
+        enemies.add(new Character("Code Crasher", 90, 20, 10));
+        enemies.add(new Character("Legacy Code Monster", 70, 40, 20));
+        enemies.add(new Character("Null Pointer Exception", 80, 20, 20));
+
+        //Создаю супермонстра - босса
+        Character darkOverlord = new Character("Dark Overlord", 200, 50, 20);
+
+
+        System.out.println("Общее количество созданных персонажей: " + Character.checkCharacterCount());
+        Character lastHero = null;
+        Random random = new Random();
+        while (heroes.size() > 0 && enemies.size() > 0) {
+            Character hero = heroes.get(random.nextInt(heroes.size()));
+            Character enemy = enemies.get(random.nextInt(enemies.size()));
+
+            System.out.println("----------------------------------------------------------------");
+            System.out.println("\nВыбранный герой: ");
```

Comment:
`\n` here might be an overkill as `System.out.println()` mean to print from new line anyway ( let's say it adds `\n` at the beginning implicitly)

`\n` would be needed if you used `System.out.print()`